tests/unit/build_providers/lxd/test_lxd.py

```diff
@@ -246,6 +245,7 @@ def test_create(self):
         ]
 
         assert check_output_calls == [
+            ["hostname", "-F", "/etc/hostname"],
```

Comment:
I think we should have a test that verifies if the hostname is actually being set inside the container.

Reply:
We can use the already existing --use-lxd test and have an override-<something> perform a check

Reply:
The existing test uses make-hello, so I added a new spread test.